reana_commons/snakemake.py

```diff
@@ -42,14 +53,82 @@ def _create_snakemake_workflow(snakefile, configfiles=None):
         configfiles = list(map(os.path.abspath, configfiles))
-        snakemake_workflow = Workflow(
+        workflow = Workflow(
             snakefile=snakefile,
             overwrite_configfiles=configfiles,
             overwrite_config=overwrite_config,
         )
 
-        snakemake_workflow.include(snakefile=snakefile)
-        return snakemake_workflow
+        workflow.include(snakefile=snakefile, overwrite_first_rule=True)
+        workflow.check()
+
+        # code copied and adapted from `snakemake.workflow.Workflow.execute()`
+        # in order to build the DAG and calculate the job dependencies.
+        # https://github.com/snakemake/snakemake/blob/75a544ba528b30b43b861abc0ad464db4d6ae16f/snakemake/workflow.py#L525
+        def rules(items):
+            return map(workflow._rules.__getitem__, filter(workflow.is_rule, items),)
+
+        if kwargs.get("keep_target_files"):
+
+            def files(items):
+                return filterfalse(workflow.is_rule, items)
+
+        else:
+
+            def files(items):
+                relpath = (
+                    lambda f: f
+                    if os.path.isabs(f) or f.startswith("root://")
+                    else os.path.relpath(f)
+                )
+                return map(relpath, filterfalse(workflow.is_rule, items))
+
+        if not kwargs.get("targets"):
+            targets = (
+                [workflow.first_rule] if workflow.first_rule is not None else list()
+            )
+
+        if kwargs.get("prioritytargets") is None:
+            prioritytargets = list()
+        if kwargs.get("forcerun") is None:
+            forcerun = list()
+        if kwargs.get("until") is None:
+            until = list()
+        if kwargs.get("omit_from") is None:
+            omit_from = list()
+
+        priorityrules = set(rules(prioritytargets))
+        priorityfiles = set(files(prioritytargets))
+        forcerules = set(rules(forcerun))
+        forcefiles = set(files(forcerun))
+        untilrules = set(rules(until))
+        untilfiles = set(files(until))
+        omitrules = set(rules(omit_from))
+        omitfiles = set(files(omit_from))
+
+        targetrules = set(
+            chain(
+                rules(targets),
+                filterfalse(Rule.has_wildcards, priorityrules),
+                filterfalse(Rule.has_wildcards, forcerules),
+                filterfalse(Rule.has_wildcards, untilrules),
+            )
+        )
+        targetfiles = set(chain(files(targets), priorityfiles, forcefiles, untilfiles))
+        dag = DAG(
+            workflow,
+            workflow.rules,
+            targetrules=targetrules,
+            targetfiles=targetfiles,
+            omitfiles=omitfiles,
+            omitrules=omitrules,
+        )
+
+        workflow.persistence = Persistence(dag=dag)
+        dag.init()
+        dag.update_checkpoint_dependencies()
+        dag.check_dynamic()
+        return dag
 
     configfiles = [kwargs.get("input")]
```

Comment:
*nitpick*: in `snakemake(configfiles=...)`, `configfiles` input is the same as `configfiles` variable here.